spec/unit/hanami/cli/commands/app/generate/struct_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+RSpec.describe Hanami::CLI::Commands::App::Generate::Struct, :app do
+  subject { described_class.new(fs: fs, inflector: inflector, out: out) }
+
+  let(:out) { StringIO.new }
+  let(:fs) { Hanami::CLI::Files.new(memory: true, out: out) }
+  let(:inflector) { Dry::Inflector.new }
+  let(:app) { Hanami.app.namespace }
+  let(:dir) { inflector.underscore(app) }
+
+  def output
+    out.rewind && out.read.chomp
+  end
+
+  context "generating for app" do
+    it "generates a struct without a namespace" do
+      subject.call(name: "book")
+
+      struct_file = <<~EXPECTED
+        # frozen_string_literal: true
+
+        module Test
+          module Structs
+            class Book < Test::DB::Struct
+            end
+          end
+        end
+      EXPECTED
+
+      expect(fs.read("app/structs/book.rb")).to eq(struct_file)
+      expect(output).to include("Created app/structs/book.rb")
+    end
+
+    it "generates a struct in a deep namespace with default separator" do
+      subject.call(name: "book.book_draft")
+
+      struct_file = <<~EXPECTED
+        # frozen_string_literal: true
+
+        module Test
+          module Structs
+            module Book
+              class BookDraft < Test::DB::Struct
+              end
+            end
+          end
+        end
+      EXPECTED
+
+      expect(fs.read("app/structs/book/book_draft.rb")).to eq(struct_file)
+      expect(output).to include("Created app/structs/book/book_draft.rb")
+    end
+
+    it "generates an struct in a deep namespace with slash separators" do
+      subject.call(name: "book/published_book")
+
+      struct_file = <<~EXPECTED
+        # frozen_string_literal: true
+
+        module Test
+          module Structs
+            module Book
+              class PublishedBook < Test::DB::Struct
+              end
+            end
+          end
+        end
+      EXPECTED
+
+      expect(fs.read("app/structs/book/published_book.rb")).to eq(struct_file)
+      expect(output).to include("Created app/structs/book/published_book.rb")
+    end
+  end
+
+  context "generating for a slice" do
+    it "generates a struct in a top-level namespace, with recommendation" do
```

Comment:
```suggestion
    it "generates a struct in a top-level namespace" do
```

I think "with recommendation" might just be a hangover from the operation generator tests?

Reply:
Busted! Fixed this